docs/guides/implementation-guides/how-to-implement-dependency-injection.md

```diff
@@ -3,13 +3,149 @@ id: how-to-implement-dependency-injection
 title: How To Implement Dependency Injection
 ---
 
-import MvvmArchitectureDiagram from '/img/guides/implementation-guides/mvvm-architecture.png';
+# How To Implement Dependency Injection in Avalonia
 
-# How To Implement Dependency Injection
-
-Content in preparation.
+[Dependency injection (DI)](https://en.wikipedia.org/wiki/Dependency_injection) allows developers to write cleaner, more modular, and testable code. It accomplishes this by creating discrete services that are passed around/created as needed. It is especially useful when you are developing multi-platform applications that need specific service implementations for each platform.
 
 This guide will show you how to use Dependency Injection (DI) with _Avalonia UI_ and the MVVM pattern. 
 
-<img src={MvvmArchitectureDiagram} alt=""/>
+Let's assume that your MainViewModel has a dependency on IMyService. The constructor for ViewModel should looks like this:
+
+```csharp
+private readonly IMyService _myService;
+
+public MainViewModel(IMyService myService)
+{
+    _myService = myService;
+}
+```
+
+Typically you would directly instantiate `MyService` and pass it into `MainViewModel`, like this:
+
+```csharp
+var window = new MainWindow
+{
+    // Note: we assume that MyService implements IMyService
+    DataContext = new MainViewModel(new MyService())
+}
+```
+
+ This works great for simple constructors that are not used very often and don't change. But this technique does not scale very well because:
+- The more dependencies your constructor has the more things you will need to instantiate and pass in. Instantiating the dependencies locally (such as by doing new MainViewModel(new MyService())) results in direct rigid coupling to a specific instance of the dependencies.
+- Similarly if MainViewModel creates it's dependencies itself (such as in the constructor body) it also becomes directly coupled to the creation of the dependencies which can result in mostly the same problems. 
```

Comment:
II probably should have proof read this more before I shared it. But it should be:

"its" not "it's". It also probably needs a comma after 
> (such as in the constructor body)